controller/src/beerocks/master/db/db.cpp

```diff
@@ -2418,6 +2479,431 @@ const std::list<sChannelScanResults> &db::get_channel_scan_results(const sMacAdd
+        return false;
+    }
+
+    LOG(DEBUG) << __FUNCTION__ << ", initial_radio = " << initial_radio_mac;
+
+    auto timestamp = std::chrono::steady_clock::now();
+    if (save_to_persistent_db) {
+        LOG(DEBUG) << __FUNCTION__
+                   << ", configuring persistent-db, initial_radio = " << initial_radio_mac;
+
+        std::unordered_map<std::string, std::string> values_map;
+        values_map.at(timestamp_str)     = timestamp_to_string_seconds(timestamp);
+        values_map.at(initial_radio_str) = tlvf::mac_to_string(initial_radio_mac);
+
+        // update the persistent db
+        if (!update_client_entry_in_persistent_db(mac, values_map)) {
+            LOG(ERROR) << "failed to update client entry in persistent-db to for " << mac;
+            return false;
+        }
+    }
+
+    n->client_initial_radio        = initial_radio_mac;
+    n->client_parameters_last_edit = timestamp;
+
+    return true;
+}
+
+sMacAddr db::get_client_initial_radio(const sMacAddr &mac)
+{
+    auto n = get_node(mac);
+    if (!n) {
+        LOG(ERROR) << "node not found for mac " << mac;
+        return network_utils::ZERO_MAC;
+    } else if (n->get_type() != beerocks::TYPE_CLIENT) {
+        LOG(ERROR) << __FUNCTION__ << "node " << mac << " is not a valid client!";
+        return network_utils::ZERO_MAC;
+    }
+
+    return n->client_initial_radio;
+}
+
+bool db::set_client_stay_on_selected_band(const sMacAddr &mac, bool stay_on_selected_band,
+                                          bool save_to_persistent_db)
+{
+    auto n = get_node(mac);
+    if (!n) {
+        LOG(ERROR) << "node not found for mac " << mac;
+        return false;
+    } else if (n->get_type() != beerocks::TYPE_CLIENT) {
+        LOG(ERROR) << __FUNCTION__ << "node " << mac << " is not a valid client!";
+        return false;
+    }
+
+    LOG(DEBUG) << __FUNCTION__ << ", stay_on_selected_band = " << stay_on_selected_band;
+
+    auto timestamp = std::chrono::steady_clock::now();
+    if (save_to_persistent_db) {
+        LOG(DEBUG) << __FUNCTION__ << ", configuring persistent-db, selected_band_enable = "
+                   << stay_on_selected_band;
+
+        std::unordered_map<std::string, std::string> values_map;
+        values_map.at(timestamp_str)            = timestamp_to_string_seconds(timestamp);
+        values_map.at(selected_band_enable_str) = std::to_string(stay_on_selected_band);
+
+        // update the persistent db
+        if (!update_client_entry_in_persistent_db(mac, values_map)) {
+            LOG(ERROR) << "failed to update client entry in persistent-db to for " << mac;
+            return false;
+        }
+    }
+
+    n->client_stay_on_selected_band =
+        (stay_on_selected_band) ? ePersistentParamBool::ENABLE : ePersistentParamBool::DISABLE;
+    n->client_parameters_last_edit = timestamp;
+
+    return true;
+}
+
+ePersistentParamBool db::get_client_stay_on_selected_band(const sMacAddr &mac)
+{
+    auto n = get_node(mac);
```

Comment:
```suggestion
    auto node = get_node(mac);
```

Reply:
replaced all new-code occurrences of `n` to `node`